tests/test_workdir.py

```diff
@@ -57,12 +59,53 @@ def test_directory_structure_generated(self, spatial_directory_list):
             assert path.exists()
 
 
-class TestCleanDirectoryTree:
-    """Tests for `clean_directory_tree()`."""
+class TestCleanFiles:
+    """Tests for `clean_directory_tree()` and `clean_cwd_logfiles)."""
 
-    @pytest.mark.parametrize("test_path", [Path("runs"), Path("src")])
-    def test_clean_directory_tree(self, test_path):
-        """Success case: directory tree does not exist after clean."""
-        test_path.mkdir()
-        clean_directory_tree()
-        assert not test_path.exists()
+    @pytest.fixture(autouse=True)
+    def set_internal_cwd(self, monkeypatch):
+        """Sets internal.CWD to pytest's working directory."""
+        monkeypatch.setattr(internal, "CWD", Path.cwd())
+
+    @pytest.fixture()
+    def src_path(self) -> Path:
+        return Path("src")
+
+    @pytest.fixture()
+    def runs_path(self) -> Path:
+        return Path("runs")
+
+    @pytest.fixture(params=["rev_number-0.log", "rev_number-200.log"])
+    def revision_log_file(self, request) -> Path:
+        return Path(request.param)
+
+    @pytest.fixture(
+        params=["benchmark_cable_qsub.sh.o21871", "benchmark_cable_qsub.sh"]
+    )
+    def pbs_job_file(self, request) -> Path:
+        return Path(request.param)
+
+    def test_clean_realisation_files(
+        self, src_path: Path, tmp_path: Path, revision_log_file: Path
+    ):
+        """Success case: Realisation files created by benchcab are removed after clean."""
+        src_path.mkdir()
+        cable_symlink = src_path / "main"
+        # tmp_path contains the path being symlinked
+        local_cable_src_path = tmp_path / "CABLE"
+        local_cable_src_path.mkdir()
+        cable_symlink.symlink_to(local_cable_src_path)
+        revision_log_file.touch()
```

Comment:
This shouldn't be here, it should be in fixture. `test_clean_realisation_files` should get the following arguments: (self, src_path, local_cable_src_path, revision_log_file) since that's the variables used to check if the test passes or fails.

Possible way to go about it for the fixtures (untested, typos possible):

```
@pytest.fixture
def local_cable_src_path(self, tmp_path) -> Path:
     local_cable_src_path = tmp_path / "CABLE"
     local_cable_src_path.mkdir()
     return local_cable_src_path

@pytest.fixture
def src_path_with_local(self, local_cable_src_path) -> Path:
        src_path = Path("src")
        src_path.mkdir()
        cable_symlink = src_path / "main"
        cable_symlink.symlink_to(local_cable_src_path)
        return src_path
```
And `revision_log_file.touch` should go in the fixture for that file.